rmf_task_ros2/src/rmf_task_ros2/Dispatcher.cpp

```diff
@@ -474,9 +479,20 @@ class Dispatcher::Implementation
       }
 
       const auto& task_request_json = msg_json["request"];
-      const std::string task_id =
+      std::string task_id =
         task_request_json["category"].get<std::string>()
-        + ".dispatch-" + std::to_string(task_counter++);
+        + ".dispatch-";
+
+      if (use_timestamp_for_task_id)
+      {
+        task_id += std::to_string(
+          std::chrono::duration_cast<std::chrono::seconds>(
```

Comment:
Actually, why do we need the duration_cast? `time_since_epoch().count()` would directly return the nanoseconds since the start.
Also, should we use node time here? (`node->get_clock()->now()`)